pymemcache/client/base.py

```diff
@@ -291,6 +293,12 @@ def _connect(self):
             if self.no_delay and sock.family == self.socket_module.AF_INET:
                 sock.setsockopt(self.socket_module.IPPROTO_TCP,
                                 self.socket_module.TCP_NODELAY, 1)
+
+            if self.tls_context is not None and self.socket_module.AF_INET:
+                sock = self.tls_context.wrap_socket(
+                    sock, server_hostname=self.server[0],
+                    do_handshake_on_connect=False,
```

Comment:
I thought `do_handshake_on_connect=False` requires us to call `SSLSocket.do_handshake()` ourselves, but I don't see the code doing that anywhere.

Reply:
I'll try moving the wrapping before the connection and removing that param. I guess it probably does the handshake in the first write/read to the socket if not connected yet.

Reply:
I _think_ we'd want to wrap on `connect`. That's the behavior I'd expect from the connection pools, at least, where we want to pay the cost of establishing the connection upfront rather than on first use.